packages/backend/src/telemetry/service.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  Counter,
+  DiagConsoleLogger,
+  DiagLogLevel,
+  MetricOptions,
+  diag,
+  metrics
+} from '@opentelemetry/api'
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-grpc'
+import { Resource } from '@opentelemetry/resources'
+import {
+  MeterProvider,
+  PeriodicExportingMetricReader
+} from '@opentelemetry/sdk-metrics'
+
+import { ConvertError, RatesService } from '../rates/service'
+import { ConvertOptions } from '../rates/util'
+import { BaseService } from '../shared/baseService'
+
+export interface TelemetryService {
+  getOrCreateMetric(name: string, options?: MetricOptions): Counter
+  getServiceName(): string | undefined
+  getBaseAssetCode(): string
+  getBaseScale(): number
+  convertAmount(
+    convertOptions: Omit<ConvertOptions, 'exchangeRate' | 'destinationAsset'>
+  ): Promise<bigint | ConvertError>
+}
+
+interface TelemetryServiceDependencies extends BaseService {
+  serviceName: string
+  collectorUrls: string[]
+  exportIntervalMillis?: number
+  aseRatesService: RatesService
+  fallbackRatesService: RatesService
+  baseAssetCode: string
+  baseScale: number
+}
+
+export function createTelemetryService(
+  deps: TelemetryServiceDependencies
+): TelemetryService {
+  return new TelemetryServiceImpl(deps)
+}
+
+class TelemetryServiceImpl implements TelemetryService {
+  private serviceName: string
+  private meterProvider?: MeterProvider
+  private fallbackRatesService: RatesService
+  private aseRatesService: RatesService
+
+  private counters = new Map()
+  constructor(private deps: TelemetryServiceDependencies) {
+    diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG)
+    this.serviceName = deps.serviceName
+    this.fallbackRatesService = deps.fallbackRatesService
+    this.aseRatesService = deps.aseRatesService
+
+    if (
+      deps.collectorUrls &&
+      Array.isArray(deps.collectorUrls) &&
+      deps.collectorUrls.length === 0
+    ) {
```

Comment:
small thing but maybe we can just do `if(!deps.collectorUrls.length){`? we know from TS at this point that it's `string[]`.

Reply:
right. thanks